internal/project/projectcollectionbuilder.go

```diff
@@ -535,7 +535,7 @@ func (b *projectCollectionBuilder) findOrCreateDefaultConfiguredProjectWorker(
 				// For composite projects, we can get an early negative result.
 				// !!! what about declaration files in node_modules? wouldn't it be better to
 				//     check project inclusion if the project is already loaded?
-				if !config.MatchesFileName(fileName) {
+				if _, ok := config.FileNamesByPath()[path]; !ok {
```

Comment:
This was a drive-by fix, but `MatchesFileName` is a more expensive function that speculates about files that may not exist; the usage here was for a file that already exists and is either properly included by the config or not.